src/helpers/http/prss_exchange_protocol.rs

```diff
@@ -0,0 +1,81 @@
+use crate::{
+    helpers::{messaging::Message, MESSAGE_PAYLOAD_SIZE_BYTES},
+    protocol::Substep,
+};
+use std::io::ErrorKind;
+use x25519_dalek::PublicKey;
+
+pub struct PrssExchangeStep;
+
+impl AsRef<str> for PrssExchangeStep {
+    fn as_ref(&self) -> &str {
+        "prss_exchange"
+    }
+}
+
+impl Substep for PrssExchangeStep {}
+
+#[derive(Debug)]
+pub struct PublicKeyChunk([u8; MESSAGE_PAYLOAD_SIZE_BYTES]);
+
+impl PublicKeyChunk {
+    pub fn chunks(pk: PublicKey) -> [PublicKeyChunk; 4] {
+        let pk_bytes = pk.to_bytes();
+
+        // These assumptions are necessary for ser/de to work
+        assert_eq!(MESSAGE_PAYLOAD_SIZE_BYTES, 8);
+        assert_eq!(pk_bytes.len(), 32);
+
+        let mut chunks = Vec::with_capacity(4);
```

Comment:
ok, I'm having a bit of trouble figuring out an alternative.

I tried doing `(0..4).map(...)`, to create an iterator of `PublicKeyChunk`s, but I can't collect them into the array directly. I'd have to go through a `Vec` first.

I tried initializing the array, but it doesn't quite make sense to set some zero value for a `PublicKeyChunk`.

Do you know a different way to avoid the `Vec`?

Reply:
ok, we talked and yes, you can collect into `ArrayVec`